### PR TITLE
Week 5 natalia diaz

### PR DESCRIPTION
**Pull request by nataliadiaz2**

### CODE REVIEW

File: code-challenges.test.js [hunk 1]
```diff
@@ -15,42 +15,137 @@
...
+        // Expected output: "L4ck4d41s1c4l"
+        const secretCodeWord2 = "Gobbledygook"
+        // Expected output: "G0bbl3dyg00k"
+        const secretCodeWord3 = "Eccentric"
+        // Expected output: "3cc3ntr1c"
+      expect(codedMessage(secretCodeWord1)).toEqual("L4ck4d41s1c4l")
+      expect(codedMessage(secretCodeWord2)).toEqual("G0bbl3dyg00k")
+      expect(codedMessage(secretCodeWord3)).toEqual("3cc3ntr1c")
+    })
+  })
```
Comment (elysemontano) [inline code-challenges.test.js:30]: This looks amazing!!

File: code-challenges.test.js [hunk 2]
```diff
@@ -15,42 +15,137 @@
...
+//   ● codedMessage › takes in a string and returns a coded message. The coded message converts 'a' to 4, 'e' to 3, 'i' to 1, and 'o' to 0.
+
+//     ReferenceError: codedMessage is not defined
 
 // b) Create the function that makes the test pass.
 
+//Create a function - codedMessage
+//Parameter - string 
+//Go through string recognizing vowels 
+//Use the replace method to replace lowercase and uppercase vowel with given number. Use g to have the replace go through the string.
```
Comment (elysemontano) [inline code-challenges.test.js:41]: Would really like to see you use the terminology used for the expression (/a|A/g, 3) in your pseudocode.  While I love this approach, I want to make sure you understand all of the concepts used in this function.

File: code-challenges.test.js [hunk 3]
```diff
@@ -15,42 +15,137 @@
...
+//Parameter - array, string 
+//Filter through array to identify if the letter is there
+//Add a logical or operator to include if the letter is uppercase
+//Return array containing the letter
+
+const wordReturn = (array, letter) => {
+   return array.filter(value => { 
+       return value.includes(letter) || value.includes(letter.toUpperCase()) 
+   })
+}
```
Comment (elysemontano) [inline code-challenges.test.js:90]: Awesome work here!  While our example letter inputs are lower cased, it is always good to think of any edge case as a stretch.  What would happen if you got an upper cased letter as the input variable?

File: code-challenges.test.js [hunk 4]
```diff
@@ -15,42 +15,137 @@
...
+        // Expected output: true
+        const hand2 = [5, 5, 3, 3, 4]
+        // Expected output: false
+        const hand3 = [5, 5, 5, 5, 4]
+        // Expected output: false
+      expect(fullHouse(hand1)).toEqual(true)
+      expect(fullHouse(hand2)).toEqual(false)
+      expect(fullHouse(hand3)).toEqual(false)
+    })
+  })
```
Comment (elysemontano) [inline code-challenges.test.js:115]: Your tests are great!

File: code-challenges.test.js [hunk 5]
```diff
@@ -15,42 +15,137 @@
...
+        if (dupe.length < 3){      
+                return false
+        } else if (dupe[0] === dupe[1] && dupe[1] !== dupe[2]) {
+                return true
+        } else if  (dupe[0] !== dupe[1] && dupe[1] === dupe[2]) {
+               return true
+        } else {
+                return false
+        }
+}
```
Comment (elysemontano) [inline code-challenges.test.js:144]: 💯

File: interview-questions.md [hunk 6]
```diff
@@ -7,53 +7,53 @@ Then, research the question to expand on your answer. Even if you feel you have
 
 1. What is a hash in Ruby?
 
-  Your answer:
+  Your answer: A hash in Ruby is a container of data set in a key:value pair. 
 
-  Researched answer:
+  Researched answer: In Ruby a hash is a container/collection of key: value pairs, with the value able to be referenced by its unique key.
```
Comment (elysemontano) [inline interview-questions.md]: While this is correct, I would really like to see you elaborate or expand further on this in the future.  One sentence answers will not be a great representation of your knowledge in an interview.

File: interview-questions.md [hunk 7]
```diff
@@ -7,53 +7,53 @@ Then, research the question to expand on your answer. Even if you feel you have
...
 
 
 
 5. What are primary keys? Why are they important?
 
-  Your answer:
+  Your answer: Primary keys are unique indentifiers, usually used in databases to track each individual piece of data to be able to refer back. Comes in handy when pieces of data are similar, the primary keys will be different. 
 
-  Researched answer:
+  Researched answer: Primary keys are unique id's that are given to every row of a table in Rails. If there is duplicate information in a table, the primary key or unique id tells the row apart by their id.
```
Comment (elysemontano) [inline interview-questions.md]: While your answers are well worded in this section, I would love to see some more on how you have interacted with these concepts as well.

File: ruby-challenges.rb [hunk 8]
```diff
@@ -10,6 +10,18 @@
...
+# Filter through the array using the .select method 
+# Determine if the value in the array includes the letter using the .includes? to determine if it is true
+# Return the array with the elements containing the letter 
+
+def returned (array, letter) 
+    array.select { |value| value.include?(letter) }
+end 
+
+p returned beverages_array, letter_o
+p returned beverages_array, letter_t
```
Comment (elysemontano) [inline ruby-challenges.rb:24]: Wonderful!

File: ruby-challenges.rb [hunk 9]
```diff
@@ -19,16 +31,73 @@
...
+            "The #{@model} bike has #{@wheels} wheels and is going #{@current_speed} mph"
+        end 
+        def pedal_faster new_speed
+            @current_speed += new_speed
+        end 
+        def brake brake_speed
+            @current_speed = @current_speed - brake_speed
+            if @current_speed.positive?
+                @current_speed 
+            elsif  @current_speed.negative?
```
Comment (elysemontano) [inline ruby-challenges.rb:69]: Nice!!!  Great use of those Ruby built-in methods 👏

**Review by elysemontano — APPROVED** [review]
Outstanding work on your coding challenges!  I would work on expanding a little more in your interview answers.  One sentence answers are the bare minimum and would be very unconversational in an interview.  Relating to things you have encountered or how you can compare the concept to something you know is a really great way to keep it conversational.
